packages/components/src/common/utils.ts

```diff
@@ -4,3 +4,16 @@ export const toKebabCase = (str: string) =>
   str.replace(/[\s_]+/g, "-").toLowerCase();
 
 export const EMPTY_OBJECT = {};
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+type PropsObject = Record<string, any>;
+
+export const filterProps = <T extends PropsObject>(
```

Comment:
great utility function 🤩 !

I asked ChatGPT for a slightly different approach to cut down runtime and enhanced typing. What do you think?

```ts
type Props = { [key: string]: any };

function filterProps<T extends Props, K extends keyof T>(
  props: T,
  excludeProps: K[]
): Partial<T> {
  const result: Partial<T> = {};

  for (const key of Object.keys(props) as (keyof T)[]) {
    if (!excludeProps.includes(key as K)) {
      result[key] = props[key];
    }
  }

  return result;
}

// example

// Define propsToExclude with specific key types rather than string[]
const allProps = { name: "John", age: 30, gender: "male" };
const propsToExclude: (keyof typeof allProps)[] = ["gender"]; // Correctly typed as array of keys from allProps

const newProps = filterProps(allProps, propsToExclude);
console.log(newProps); // Output: { name: 'John', age: 30 }
```

Reply:
Woow, much better! Thanks for optimizing this 🥳